ext/opcache/jit/zend_jit_arm64.dasc

```diff
@@ -15093,6 +15122,55 @@ static int zend_jit_add_veneer(dasm_State *Dst, void *buffer, uint32_t ins, int
 				return n;
 			}
 		}
+	} else if (JIT_G(trigger) == ZEND_JIT_ON_HOT_TRACE
+	 && (ins >> 16) == DASM_REL_A) {
+		ptrdiff_t addr = (((ptrdiff_t)(*(b-1))) << 32) | (unsigned int)(*(b-2));
+
+		if ((void*)addr >= dasm_buf && (void*)addr < dasm_end) {
+			uint32_t exit_point = zend_jit_trace_find_exit_point((void*)addr);
+			zend_jit_trace_info *t = zend_jit_get_current_trace_info();
+
+			if (exit_point != (uint32_t)-1) {
+				/* Use exit points table */
+
+				ZEND_ASSERT(exit_point < t->exit_count);
+
+				veneer = (char*)buffer + dasm_getpclabel(&Dst, 1) - (t->exit_count - exit_point) * 4;
+				na = (ptrdiff_t)veneer - (ptrdiff_t)cp + 4;
+				n = (int)na;
+
+				/* check if we can jump to veneer */
```

Comment:
These checks are almost duplicate to lines 15185 to 15216, except the missing check at line 15210.
I'm not sure whether the check at lien 15210 is missing intentionally or not.
Besides, if they are duplicate, do you think we should extract them and put them into one common function?

Reply:
Based on my understanding, the question I asked in https://github.com/php/php-src/commit/fc3157ea0f, is not an issue here because the **exit address** is usually far from `cp` whereas our `veneer` is put just at the end of JIT-ed trace code and near to `cp`, right?

Reply:
This exit table is similar to veneers. The only difference that target lables may be patched.

Of course, it's possible to improve the code. Extracting common code into function makes sense.